kotest-property-arrow/src/commonMain/kotlin/io/kotest/property/arrow/laws/Law.kt

```diff
@@ -2,27 +2,26 @@ package io.kotest.property.arrow.laws
 
 import io.kotest.assertions.fail
 import io.kotest.core.names.TestName
-import io.kotest.core.spec.style.scopes.RootScope
-import io.kotest.core.spec.style.scopes.addTest
+import io.kotest.core.spec.style.scopes.ContainerScope
 import io.kotest.core.test.TestScope
 
 public data class Law(val name: String, val test: suspend TestScope.() -> Unit)
 
 public fun <A> A.equalUnderTheLaw(other: A, f: (A, A) -> Boolean = { a, b -> a == b }): Boolean =
   if (f(this, other)) true else fail("Found $this but expected: $other")
 
-public fun RootScope.testLaws(vararg laws: List<Law>): Unit =
+public suspend fun ContainerScope.testLaws(vararg laws: List<Law>): Unit =
   laws
     .flatMap { list: List<Law> -> list.asIterable() }
     .distinctBy { law: Law -> law.name }
     .forEach { law ->
-      addTest(TestName(law.name), disabled = false, config = null) { law.test(this) }
+      registerTest(TestName(law.name), disabled = false, config = null) { law.test(this) }
     }
 
-public fun RootScope.testLaws(prefix: String, vararg laws: List<Law>): Unit =
+public suspend fun ContainerScope.testLaws(prefix: String, vararg laws: List<Law>): Unit =
   laws
     .flatMap { list: List<Law> -> list.asIterable() }
     .distinctBy { law: Law -> law.name }
     .forEach { law: Law ->
-      addTest(TestName(prefix, law.name, true), disabled = false, config = null) { law.test(this) }
```

Comment:
@i-walker this `addTest` is no longer allowed after spec is instantiated https://github.com/kotest/kotest/pull/2872

Reply:
It never was, it just silently ignored the test. So in 5.2 it actually lets you know it's broken.

Reply:
@i-walker I believe under normal circumstances we normally do that within container scope. I tried fixing that here.

Reply:
Awesome thank you I didn‘t know the context :)